File: Models/Book.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace NgLibrary.Models
{
    public class Book : IRentable
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key] public string Id { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string CoverImg { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public string Author { get; set; } = string.Empty;
        public string Publisher { get; set; } = string.Empty;
        public DateTime? PublishedDate { get; set; }
        public string Category { get; set; } = string.Empty;
        public string ISBN { get; set; } = string.Empty;
        public int PageCount { get; set; }
        public int Available { get; set; }
        public int TotalCount { get; set; } = 1;
    }
}

File: NgLibraryUI/src/app/models/book.ts
import { IRentable } from "./rentable";

export class Book implements IRentable {
    id: string = '';
    description: string ='';
    coverImg: string = '';
    title: string = '';
    author: string = '';
    publisher: string = '';
    publishedDate: Date = new Date();
    category: string = '';
    isbn: string = '';
    pageCount: number = 0;
    available: number = 0;
    totalCount: number = 1;
}